packages/fxa-settings/src/components/UnitRowSecondaryEmail/index.tsx

```diff
@@ -1,80 +1,235 @@
-            <p>Check the inbox for {primaryEmail} to verify your primary email.</p>
+            {data && (
+              <p>
+                Check the inbox for {primaryEmail} to verify your secondary
+                email.
+              </p>
+            )}
+            {error && <p>Error text TBD. {error.message}</p>}
           </AlertBar>
         )}
-      </UnitRow>
+        <div className="mobileLandscape:flex unit-row-multi-row">
+          <div className="unit-row-content" data-testid="unit-row-content">
+            <p className="font-bold" data-testid="unit-row-header-value">
+              {secondaryEmail}
+              {!secondaryEmailIsVerified && (
+                <span
+                  data-testid="unverified-text"
+                  className="uppercase block text-orange-600 font-bold text-xs"
+                >
+                  unverified
+                </span>
+              )}
+            </p>
+            {/* Only include SecondaryEmailDefaultContent on last verified secondary email */}
+            {secondaryEmailIsVerified &&
+              lastVerifiedSecondaryEmailIndex === index && (
+                <SecondaryEmailDefaultContent />
+              )}
+            {!secondaryEmailIsVerified && (
+              <p className="text-xs mt-3 text-grey-400">
+                Verification needed.
+                <button
+                  className="link-blue mx-1"
+                  data-testid="resend-secondary-email-code-button"
+                  onClick={resendSecondaryEmailCodeHandler}
+                >
+                  Resend verification email
+                </button>
+                if it's not in your email or spam.
+              </p>
+            )}
+          </div>
+          <div className="unit-row-actions" data-testid="unit-row-actions">
+            {secondaryEmailIsVerified && (
+              <div>
+                <button
+                  className="cta-neutral transition-standard"
+                  onClick={() => {
+                    console.log(`make ${secondaryEmail} the primary email!`);
+                  }}
+                  data-testid="secondary-email-make-primary"
+                >
+                  Make primary
+                </button>
+              </div>
+            )}
+          </div>
+        </div>
+      </>
+    );
+  };
+
+  if (!hasAtLeastOneSecondaryEmail) {
+    return <UnitRowSecondaryEmailNotSet />;
+  }
+
+  // user has at least one secondary email (verified or unverified) set
+  return (
+    <div className="unit-row">
+      <div className="unit-row-header">
+        <h3 data-testid="unit-row-header">Secondary email</h3>
+      </div>
+      <div className="mobileLandscape:flex-3 desktop:flex-5">
+        {secondaryEmails.map((secondaryEmailObj, index) => (
+          <UnitRowSecondaryEmailContentAndActions
+            key={secondaryEmailObj.email}
+            {...{
+              secondaryEmailObj,
+              index,
```

Comment:
Instead of the component trying to figure out if it's the last one of the bunch, could we just tell it? Then it doesn't have to do any index logic bidness

Reply:
You mean `isLastSecondaryEmail={index === lastVerifiedSecondaryEmailIndex}`? I can do that.